cpp/src/groupby/hash/groupby.cu

```diff
@@ -426,16 +498,17 @@ void compute_single_pass_aggs(table_view const& keys,
 {
   // flatten the aggs to a table that can be operated on by aggregate_row
   table_view flattened_values;
-  std::vector<aggregation::Kind> aggs;
+  std::vector<std::unique_ptr<aggregation>> aggs;
+  std::vector<aggregation::Kind> agg_kinds;
   std::vector<size_t> col_ids;
-  std::tie(flattened_values, aggs, col_ids) = flatten_single_pass_aggs(requests);
+  std::tie(flattened_values, agg_kinds, aggs, col_ids) = flatten_single_pass_aggs(requests);
```

Comment:
Would structured bindings be a good choice here?

Reply:
Very timely comment.  Mark ended up making that change in a separate PR, which I will be merging shortly.